fbpic/main.py

```diff
@@ -312,13 +326,11 @@ def step(self, N=1, correct_currents=True,
         if self.comm.size > 1 and correct_divE:
             raise ValueError('correct_divE cannot be used in multi-proc mode.')
 
-        # Let the user know that the first step is much longer
-        if show_progress and (self.comm.rank==0):
-            print('Performing %d PIC iterations:' %N )
-            print(' - Just-In-Time compilation (up to one minute) ...')
-
-        # Measure the time taken by the PIC cycle
-        measured_start = time.time()
+        # Initialize variables to measure the time taken by the simulation
```

Comment:
I like how the modifications below clean the code in the main loop. 
One possibility to make it even cleaner would be to write a class (e.g. `TimePrinter`, in `printing_utils.py`) and create an instance of that class, here, that records the `start_time`, `prev_time` and `avg_time_per_step`, so that these variables do not appear explicitly in the code for the `step` function.
Then this instance could have the method `progression_bar` and `print_runtime_summary`.

Reply:
Great idea, I created a class ProgressBar which handles all of this